modular_skyrat/master_files/code/modules/clothing/under/akula_jobs.dm

```diff
@@ -0,0 +1,207 @@
+/obj/item/clothing/proc/apply_wetsuit_status_effect(mob/living/carbon/human/user)
+	if(!HAS_TRAIT(user, TRAIT_SLICK_SKIN))
+		return FALSE
+	user.apply_status_effect(/datum/status_effect/wetsuit)
+
+/// A proc to remove the wetsuit status effect, used with the `dropped()` proc
+/obj/item/clothing/proc/remove_wetsuit_status_effect(mob/living/carbon/human/user)
+	if(!HAS_TRAIT(user, TRAIT_SLICK_SKIN))
+		return FALSE
+	user.remove_status_effect(/datum/status_effect/wetsuit)
+
+/datum/status_effect/wetsuit
+	id = "wetsuit"
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = null
+	tick_interval = 10 SECONDS
+
+/datum/status_effect/wetsuit/tick()
+	owner.set_wet_stacks(15)
+	..()
+
+/obj/item/clothing/under/akula_wetworks
+	name = "wetworks envirosuit"
+	desc = ""
+	icon_state = "default"
+	inhand_icon_state = null
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/under/akula.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/under/akula.dmi'
+	armor_type = /datum/armor/wetworks_under
+	body_parts_covered = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	can_adjust = FALSE
+	strip_delay = 40
+	/// If an akula tail accessory is present, we can overlay a stylish additional icon
+	var/style_overlay
+
+/obj/item/clothing/under/akula_wetworks/Initialize(mapload)
+	. = ..()
+	update_appearance()
+
+/obj/item/clothing/under/akula_wetworks/worn_overlays(mutable_appearance/standing, isinhands)
+	. = ..()
+	if(!isinhands && style_overlay)
+		. += mutable_appearance('modular_skyrat/master_files/icons/mob/clothing/under/akula.dmi', style_overlay, ABOVE_MOB_LAYER)
+	else
+		cut_overlays()
+
+/obj/item/clothing/under/akula_wetworks/equipped(mob/user, slot)
+	. = ..()
+	check_physique(user)
+	check_style_overlay(user)
+	apply_wetsuit_status_effect(user)
+	update_appearance()
+
+/obj/item/clothing/under/akula_wetworks/dropped(mob/user)
+	. = ..()
+	remove_wetsuit_status_effect(user)
+
+/// This will check the wearer's bodytype and change the wetsuit worn sprite according to if its male/female
+/obj/item/clothing/under/akula_wetworks/proc/check_physique(mob/living/carbon/human/user)
+	icon_state = initial(icon_state)
+	if(user.physique == FEMALE)
+		icon_state = "[icon_state]_f"
+	return TRUE
+
+/// Checks if the wearer has a compatible tail for the `style_overlay` variable
+/obj/item/clothing/under/akula_wetworks/proc/check_style_overlay(mob/living/carbon/human/user)
+	// No tail
+	if(!istype(user.getorganslot(ORGAN_SLOT_EXTERNAL_TAIL), /obj/item/organ/external/tail))
+		return FALSE
+
+	var/tail = user.dna.species.mutant_bodyparts["tail"][MUTANT_INDEX_NAME]
+	switch(tail)
+		if("Akula")
+			style_overlay = "overlay_akula"
+		if("Shark")
+			style_overlay = "overlay_shark"
+		if("Shark (No Fin)")
+			style_overlay = "overlay_shark_no_fin"
+		if("Fish")
+			style_overlay = "overlay_fish"
+		else
+			style_overlay = null
+
+	/// Suit armor
+/datum/armor/wetworks_under
```

Comment:
this feels pretty strong for being a _jumpsuit_

Reply:
armor values are still unedited, this is copied directly from plasmaman envirosuit and will be changed later